jetty-core/jetty-tests/jetty-test-client-transports/src/test/java/org/eclipse/jetty/test/client/transport/HttpClientStreamTest.java

```diff
@@ -1193,6 +1197,82 @@ public boolean handle(Request request, org.eclipse.jetty.server.Response respons
         assertTrue(clientLatch.await(timeoutInSeconds, TimeUnit.SECONDS));
     }
 
+    @ParameterizedTest
+    @MethodSource("transports")
+    @Tag("DisableLeakTracking:server")
+    public void testUploadWithRetainedData(Transport transport) throws Exception
+    {
+        start(transport, new Handler.Abstract()
+        {
+            @Override
+            public boolean handle(Request request, org.eclipse.jetty.server.Response response, Callback callback)
+            {
+                new Runnable()
+                {
+                    @Override
+                    public void run()
+                    {
+                        while (true)
+                        {
+                            try
+                            {
+                                Thread.sleep(100);
```

Comment:
what is this for?

Reply:
This test is about triggering a timeout on the client during an upload. This `sleep()` makes sure the server is slow enough so that the timeout triggers but fast enough so that it starts consuming some of the uploaded chunks.